test/test_string_array.cpp

```diff
@@ -130,3 +130,60 @@ TEST(test_string_array, string_array_cmp) {
   ret = rcutils_string_array_fini(&incomplete_string_array);
   ASSERT_EQ(RCUTILS_RET_OK, ret);
 }
+
+TEST(test_string_array, string_array_sort) {
+  auto allocator = rcutils_get_default_allocator();
+  rcutils_ret_t ret;
+
+  ret = rcutils_string_array_sort(nullptr);
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, ret);
+
+  rcutils_string_array_t sa0 = rcutils_get_zero_initialized_string_array();
+  ret = rcutils_string_array_sort(&sa0);
+  EXPECT_EQ(RCUTILS_RET_OK, ret);
+
+  ret = rcutils_string_array_init(&sa0, 8, &allocator);
+  ASSERT_EQ(RCUTILS_RET_OK, ret);
+
+  // Already in order
+  for (size_t i = 0; i < sa0.size; i++) {
+    const char val[] = {static_cast<char>('a' + i), '\0'};
+    sa0.data[i] = strdup(val);
```

Comment:
AFAIK, using `strdup` like this without `free` calls leads to memory leaks, if this is the case you should add those.

Reply:
you can use `rcutils_strdup`

Reply:
A call to `rcutils_string_array_fini` will free the resources of the members as well. That said, I'm not calling `rcutils_string_array_fini` like I should, so I'll add that call.

In terms of `rcutils_strdup` vs `strdup`, I would prefer to maintain a consistent pattern in this file. If you think this should change, feel free to open a PR to adjust the other two places and I'll align this change to conform.

Reply:
Test resources should be freed by acdbccd.